### PR TITLE
Display user presented under chat bar

### PR DESCRIPTION
**Pull request by leekkww**
Changes:
- displays user presented under chat bar, with reasonable wrapping if name is too long
- moves user information (color, displayname) to a single object in the reducer
- messages now show current name corresponding to each sender (instead of the sender name at the time of message sent)

### CODE REVIEW

File: src/components/Chat/index.js [hunk 1]
```diff
@@ -238,11 +259,11 @@ export default class Chat extends Component {
     const {text, senderId: id, isOpponent} = message;
     const big = text.length <= 10 && isEmojis(text);
     const color = this.getMessageColor(id, isOpponent);
+    const users = this.props.users;
 
-    this.props.updateSeenChatMessage && this.props.updateSeenChatMessage(message);
```
Comment (leekkww) [inline src/components/Chat/index.js]: this line is unrelated to this change. react was erroring saying that we shouldn't update anything while we are rendering, and the error goes away if we remove this line. lmk if you want this line back instead.

Comment (stevenhao) [reply src/components/Chat/index.js]: ah, i didn't notice this line. it should go in componentDidMount, or ideally, in a callback that triggers when the scrolls into view in the page. deleting for now is fine, i'll find a way to fix it.

**Comment by stevenhao** [comment]
closes #65
